src/main/java/uk/gov/hmcts/cmc/claimstore/services/InterestRateService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.cmc.claimstore.services;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.InterestAmount;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+
+@Component
+public class InterestRateService {
+
+    public InterestAmount calculateRate(LocalDate fromDate, LocalDate toDate, double rate, double amount) {
+        validateNumberIsNonNegative(rate);
+        validateNumberIsNonNegative(amount);
+
+        final double noOfDays = calculateNumberOfDays(fromDate, toDate);
```

Comment:
What if number of days is negative e.g somebody passed end date is before start date? We should probably reject such calculation as that indicates programmatic error.

Reply:
Please see `validatePeriod(fromDate, toDate);`

```
private double calculateNumberOfDays(final LocalDate fromDate, final LocalDate toDate) {
    validatePeriod(fromDate, toDate);
    return fromDate.until(toDate, ChronoUnit.DAYS);
}
```